BinaryProtocol-8bit.md

```diff
@@ -0,0 +1,347 @@
+      U32 = (isUnsigned | 4),
+      S32 = (isSigned   | 4),
+      U64 = (isUnsigned | 8),
+      S64 = (isSigned   | 8),
+      Float = (isFloat  | 4),
+      Timestamp = hasTimestamp,
+      TimestampedU8  = (hasTimestamp | U8),
+      TimestampedS8  = (hasTimestamp | S8),
+      TimestampedU16 = (hasTimestamp | U16),
+      TimestampedS16 = (hasTimestamp | S16),
+      TimestampedU32 = (hasTimestamp | U32),
+      TimestampedS32 = (hasTimestamp | S32),
+      TimestampedU64 = (hasTimestamp | U64),
+      TimestampedS64 = (hasTimestamp | S64),
+      TimestampedFloat = (hasTimestamp | Float)
+  }
+```
+
+The field `PayloadType` has a flag on the 5th least significant bit that indicates if the time information is available on the Harp Message. The existence of this flag is useful to know if the fields [Seconds] and [Nanoseconds] are present on the Harp Message.
+In `C` one can check if the time information is avaible by using the following snippet:
+
+```C
+int hasTimestamp = 0x10;
+
+if (PayloadType &  hasTimestamp )
+{
+    printf(“The time information is available on the Harp Message’s Payload.\n”);
+}
+```
+
+### Using [Checksum] to validate communication integrity
+
+The `Checksum` field is the sum of all bytes contained in the Harp Message. The receiver of the message should calculate the checksum and compare it with the received. If they don’t match, the Harp Message should be discarded.
+Example on how to calculate the [Checksum] in C language:
+
+```C
+unsigned char Checksum = 0;
+int i = 0;
+for (; i < Length + 1; i++ )
+{
+    Checksum += HarpMessage(i);
+}
+```
+
+### Parsing [Payload] with Arrays
+
+The `Payload` element can contain a single, or an array of values of the same type. The first step to parse these payloads is to first find the number of values contained on the [Payload] element. This can be done using the following `C` code example:
+
+```C
+int arrayLength;
+int hasTimestamp = 0x10;
+int sizeMask = 0x0F;
+
+if (PayloadType & hasTimestamp)
+{
+    // Harp Message has time information
+    arrayLength = (Length – 10) / (PayloadType & sizeMask )
+}
+else
+{
+    // Harp Message doesn’t have time information
+    arrayLength = (Length – 4) / (PayloadType & sizeMask )
+}
+```
+---
+
+## Typical usage
+
+### Commands
+
+The `Peripheral` device that runs the `Harp Protocol` receives `Write` and `Read` `Commands` from the `Host`, and, in turn, sends/responds with `Replies`.
+
+Some `Harp Message`s are shown below to demonstrate the typical usage of the protocol between a `Peripheral` and an `Host`. Note that, from the `Host` to the `Peripheral`, the time information is omitted in Harp Message, since this information is optional.
+We will use the following abbreviations:
+
+  - [CMD] is a Command (From the `Host` to the `Peripheral`);
+  - [RPL] is a Reply (From `Peripheral` to the `Host`)
+  - [EVT] is an Event. (A message sent from the `Peripheral` to the `Host` without a request from the `Host`)s
+
```

Comment:
- [EVT] is an Event. (A message sent from the `Peripheral` to the `Host` without a request from the `Host`)s
--> - [EVT] is an Event (A message sent from the `Peripheral` to the `Host` without a request from the `Host`)